apps/tangle-dapp/data/restaking/useRestakingEraReward.ts

```diff
@@ -0,0 +1,74 @@
+import { createWorkerFactory, useWorker } from '@shopify/react-web-worker';
```

Comment:
Can you elaborate more on the performance issue you are facing that require us to use web worker here?

Reply:
You can inspect the logic within the worker's functions (links below), which entail numerous queries and calculations. Therefore, I've relocated the intensive calculations here to prevent UI blocking.

- [calculateEarnings](https://github.com/webb-tools/webb-dapp/pull/2195/files#diff-3978663ae49174f42687e4d10a9b741855ec9f6c97617b9bb9dba76f50ea8f54).
- [calculateEraRewardPoint](https://github.com/webb-tools/webb-dapp/pull/2195/files#diff-6847dee29a05fa26a3a4119e86a1d39e407b995f97c1921160571383e4de1383).